### PR TITLE
add support for --tls flag (#203)

### PR DESCRIPTION
**Pull request by r-mol**
## What was changed
1. client/factory.go -> Added parsing `--tls` flag with checking on error. And added the logic that when the flag is activated, the host is taken from the address/localhost
2. common/defs-flags.go -> Added related definition for ` --tls` flag
3. common/flags.go -> Added const of `--tls` flag name. And added `--tls` flag to SharedFlag slice

## Why?
I am connecting to my server over a TLS connection and I had to duplicate my host from the `--address` flag to the `--tls_server_name` flag. This cluttered up the command, as well as misleading.

## Checklist

1. Closes [#203](https://github.com/temporalio/cli/issues/203)

4. How was this tested:
- By internal tests, but with adding this flag
- By connecting to local server with this flag

5. Any docs updates needed? - NO

## Related to same issue
https://github.com/temporalio/tctl/pull/360

### CODE REVIEW

File: client/factory.go [hunk 1]
```diff
@@ -216,6 +216,11 @@ func (b *clientFactory) createTLSConfig(c *cli.Context) (*tls.Config, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to read TLS disable host verification flag: %w", err)
 	}
+	enableTLSS := c.String(common.FlagTLS)
+	enableTLS, err := strconv.ParseBool(enableTLSS)
```
Comment (feedmeapples) [inline client/factory.go:220]: what's the reason to do custom parsing? (instead of `c.Bool(..)`)

File: common/defs-flags.go [hunk 2]
```diff
@@ -1,10 +1,12 @@
 package common
+
 const (
 	// Shared flag definitions
 	FlagEnvDefinition = "Environment to read environmental variables from."
 	FlagAddrDefinition = "The host and port (formatted as host:port) for the Temporal Frontend Service."
 	FlagNSAliasDefinition = "Identifies a Namespace in the Temporal Workflow."
 	FlagMetadataDefinition = "Contains gRPC metadata to send with requests (formatted as key=value). Values must be in a valid JSON format."
+	FlagTLSDefinition = "Enable TLS"
```
Comment (feedmeapples) [inline common/defs-flags.go:9]: we should elaborate here for users, maybe
```suggestion
	FlagTLSDefinition = "Enable TLS encryption without additional options such as mTLS or client certificates"
```

**Review by feedmeapples — CHANGES_REQUESTED** [review]

File: client/factory.go (hunk 1 above)
Comment (r-mol) [reply client/factory.go:220]: I'm doing custom parsing to keep consistency with your code above: https://github.com/temporalio/cli/blob/6a16f1c6ef215dfe88b4bb8598bd7c21bc4976c9/client/factory.go#L214

Comment (feedmeapples) [reply client/factory.go:220]: ok so these values are accepted with strconv.ParseBool, true: `"1", "t", "T", "true", "TRUE", "True"`, false: `"0", "f", "F", "false", "FALSE", "False"`

I'd go with the c.Bool for consistency with the rest of the CLI

**Review by feedmeapples — APPROVED** [review]

File: client/factory.go (hunk 1 above)
Comment (feedmeapples) [reply client/factory.go:220]: updated

File: common/defs-flags.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ const (
 	FlagAddrDefinition           = "The host and port (formatted as host:port) for the Temporal Frontend Service."
 	FlagNSAliasDefinition        = "Identifies a Namespace in the Temporal Workflow."
 	FlagMetadataDefinition       = "Contains gRPC metadata to send with requests (formatted as key=value)."
+  FlagTLSDefinition            = "Enable TLS encryption without additional options such as mTLS or client certificates"
```
Comment (feedmeapples) [inline common/defs-flags.go:9]:
```suggestion
	FlagTLSDefinition            = "Enable TLS encryption without additional options such as mTLS or client certificates"
```

**Comment by feedmeapples** [comment]
Tests fail due to recent regression unrelated to the current PR, specifically verifying RC code in `TestServerInterruptRC` and E2E tests running indefinitely. Approving and fixing the tests in different PR

**Comment by r-mol** [comment]
And could you also check the PR with the same issue https://github.com/temporalio/tctl/pull/360.

**Pull request merged**